MetaMorpheus/TaskLayer/MetaMorpheusTask.cs

```diff
@@ -165,34 +165,48 @@ public static List<Ms2ScanWithSpecificMass>[] _GetMs2Scans(MsDataFile myMSDataFi
                                     precursorSpectrum.MassSpectrum, commonParameters.PrecursorDeconvolutionParameters))
                                 {
                                     double monoPeakMz = envelope.MonoisotopicMass.ToMz(envelope.Charge);
-                                    precursors.Add((monoPeakMz, envelope.Charge));
+                                    int peakCount = envelope.Peaks.Count();
+                                    double intensity = 1;
+                                    if (commonParameters.UseMostAbundantPrecursorIntensity) 
+                                    { 
+                                        intensity = envelope.Peaks.Max(p => p.intensity); 
+                                    }
+                                    else
+                                    {
+                                        intensity = envelope.Peaks.Sum(p => p.intensity);
+                                    }
+                                    precursors.Add((monoPeakMz, envelope.Charge, intensity, peakCount));
```

Comment:
Should you also store which m/z has the highest intensity? I feel like you might want that information later